lib/views/chatroom/chatroom_components/chat_bubble.dart

```diff
@@ -0,0 +1,367 @@
+    String _time = widget.time ?? "12:00";
+    String _profileImageUrl =
+        widget.profileImageUrl ?? "https://picsum.photos/200/300";
+
+    return Column(
+      crossAxisAlignment:
+          _isSent ? CrossAxisAlignment.end : CrossAxisAlignment.start,
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        if (_showReactions)
+          Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 64.0),
+            child: ReactionBar(
+              refresh: refresh,
+              reactions: reactions,
+            ),
+          ),
+        const SizedBox(height: 8),
+        Padding(
+          padding: const EdgeInsets.symmetric(
+            horizontal: 18,
+            vertical: 4,
+          ),
+          child: Row(
+            mainAxisAlignment:
+                _isSent ? MainAxisAlignment.end : MainAxisAlignment.start,
+            crossAxisAlignment: CrossAxisAlignment.end,
+            children: [
+              !_isSent
+                  ? Container(
+                      width: 42,
+                      height: 42,
+                      decoration: BoxDecoration(
+                        color: Colors.blue,
+                        borderRadius: BorderRadius.circular(21),
+                        image: _profileImageUrl != null
+                            ? DecorationImage(
+                                image: NetworkImage(_profileImageUrl),
+                                fit: BoxFit.cover,
+                              )
+                            : null,
+                      ),
+                    )
+                  : const SizedBox(),
+              const SizedBox(width: 6),
+              CustomPopupMenu(
+                controller: _controller,
+                pressType: PressType.longPress,
+                showArrow: false,
+                verticalMargin: 12,
+                menuBuilder: () => ClipRRect(
+                  child: Container(
+                    width: getWidth(context) * 0.4,
+                    color: Colors.white,
+                    child: IntrinsicWidth(
+                      child: Column(
+                        crossAxisAlignment: CrossAxisAlignment.stretch,
+                        children: [
+                          ListTile(
+                            onTap: () {
+                              Fluttertoast.showToast(msg: "Reply to message");
+                              _controller.hideMenu();
+                            },
+                            leading: const Icon(
+                              Icons.reply_outlined,
+                              size: 18,
+                            ),
+                            title: Text(
+                              "Reply",
+                              style: GoogleFonts.roboto(
+                                fontSize: 16,
+                              ),
+                            ),
+                          ),
+                          ListTile(
```

Comment:
It will be better to breakdown this ListTile into another widget, because it is being used at multiple places.

Reply:
It will only require that many arguments, so left it like this